ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -0,0 +1,578 @@
+                                  << m_attrs.cube_coeff << "\n\n\n";
+                        cubic_func(input_data, out);
+                        break;
+                    }
+                }
+
+            private:
+                op::v4::Interpolate::InterpolateAttrs m_attrs;
+
+                InterpolateMode m_interp_mode;
+                double m_cube_coeff;
+
+                Shape m_input_data_shape;
+                std::vector<int64_t> m_axes;
+                Shape m_out_shape;
+
+                std::vector<float> m_scales;
+
+                InterpolateEvalHelper helper;
+
+                void linear_func(const T* input_data, T* out);
+                void linear_onnx_func(const T* input_data, T* out);
+                void cubic_func(const T* input_data, T* out);
+                void nearest_func(const T* input_data, T* out);
+            };
+
+            template <typename T>
+            void InterpolateEval<T>::linear_func(const T* input_data, T* out)
+            {
+                size_t input_rank = m_input_data_shape.size();
+                size_t num_of_axes = m_axes.size();
+
+                auto info = helper.get_info_for_linear_mode();
+
+                CoordinateTransform output_transform(m_out_shape);
+                CoordinateTransform input_transform(m_input_data_shape);
+
+                for (const Coordinate& output_coord : output_transform)
+                {
+                    auto icoords_data = helper.get_icoords(output_coord);
+
+                    float summa = 0.0f;
+                    float wsum = 0.0f;
+
+                    CoordinateTransform indices{info.shape_for_indeces};
+                    for (const auto& index : indices)
+                    {
+                        auto inner_result =
+                            helper.inner_calculation(output_coord, icoords_data, info, index);
+                        if (!inner_result.condition)
+                        {
+                            continue;
+                        }
+
+                        wsum += inner_result.w;
+                        summa += inner_result.w *
+                                 static_cast<float>(
+                                     input_data[input_transform.index(inner_result.inner_coord)]);
+                    }
+
+                    if (wsum == 0.0f)
+                    {
+                        out[output_transform.index(output_coord)] = T{};
+                    }
+                    else
+                    {
+                        out[output_transform.index(output_coord)] = static_cast<T>(summa / wsum);
+                    }
+                }
+            }
+
+            template <typename T>
+            void InterpolateEval<T>::linear_onnx_func(const T* input_data, T* out)
+            {
+                size_t input_rank = m_input_data_shape.size();
+
+                assert((input_rank == 2) || (input_rank == 4));
+                assert(m_axes.size() == 2);
```

Comment:
I think in some case, `m_axes.size() == 4` could be ture. e.g.: when input_rank is 4, axes could be set with full size, or not passed as optional input, so default is [0,1,2,3] with rank of 4.  am I right?

Reply:
Fixed.